lib/Doctrine/DBAL/Migrations/Version.php

```diff
@@ -463,17 +425,36 @@ private function formatParamsForOutput(array $params, array $types)
         return sprintf('with parameters (%s)', implode(', ', $out));
     }
 
-    private function dispatchEvent($eventName, $direction, $dryRun)
-    {
-        $this->configuration->dispatchEvent($eventName, new MigrationsVersionEventArgs(
+    private function dispatchEvent(
+        string $eventName,
+        string $direction,
+        bool $dryRun
+    ) : void {
+        $event = $this->createMigrationsVersionEventArgs(
+            $this,
+            $this->configuration,
+            $direction,
+            $dryRun
+        );
+
+        $this->configuration->dispatchEvent($eventName, $event);
```

Comment:
Dispatching event on Configuration is really weird, should be `@todo`ed for refactoring.

Reply:
Added issue for this here #629